packages/jest-environment-jsdom/src/index.ts

```diff
@@ -15,24 +15,17 @@ import {JSDOM, VirtualConsole} from 'jsdom';
 
 // The `Window` interface does not have an `Error.stackTraceLimit` property, but
 // `JSDOMEnvironment` assumes it is there.
-interface Win extends Window {
-  Error: {
-    stackTraceLimit: number;
+type Win = Window &
+  Global.Global & {
+    Error: {
+      stackTraceLimit: number;
+    };
   };
-}
-
-function isWin(globals: Win | Global.Global): globals is Win {
-  return (globals as Win).document !== undefined;
-}
-
-// A lot of the globals expected by other APIs are `NodeJS.Global` and not
-// `Window`, so we need to cast here and there
 
 class JSDOMEnvironment implements JestEnvironment {
   dom: JSDOM | null;
   fakeTimers: FakeTimers<number> | null;
-  // @ts-ignore
-  global: Global.Global | Win | null;
+  global: Win;
```

Comment:
If this keeps the `| null` the ts-ignore in teardown is not necessary, right?

Reply:
If we add `| null` to it, Typescript will skip ignoring when creating types. We just (or should) add `@ts-ignore` in code flow in this case!

Reply:
Oh the ignore was there because `JestEnvironment` doesn't allow it to be null, now I understand ^^